pkg/config/config.go

```diff
@@ -248,9 +246,28 @@ func (c *MicroshiftConfig) ReadAndValidate(configFile string, flags *pflag.FlagS
 		return err
 	}
 
+	// validate serviceCIDR
+	_, err := GetClusterDNS(c.Cluster.ServiceCIDR)
```

Comment:
Why not go ahead and store the value in the struct here instead of making the caller compute it again and handle the error separately? GetClusterDNS could be a private function.